File: app/java/src/main/java/com/example/production/domain/model/purchase/PurchaseOrderDetail.java
package com.example.production.domain.model.purchase;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PurchaseOrderDetail {
    private Integer id;
    private String purchaseOrderNumber;
    private Integer lineNumber;
    private String orderNumber;
    private String deliveryLocationCode;
    private String itemCode;
    @Builder.Default
    private Boolean miscellaneousItemFlag = false;
    private LocalDate expectedReceivingDate;
    private LocalDate confirmedDeliveryDate;
    private BigDecimal orderUnitPrice;
    private BigDecimal orderQuantity;
    @Builder.Default
    private BigDecimal receivedQuantity = BigDecimal.ZERO;
    @Builder.Default
    private BigDecimal inspectedQuantity = BigDecimal.ZERO;
    @Builder.Default
    private BigDecimal acceptedQuantity = BigDecimal.ZERO;
    private BigDecimal orderAmount;
    @Builder.Default
    private BigDecimal taxAmount = BigDecimal.ZERO;
    @Builder.Default
    private Boolean completedFlag = false;
    private String detailRemarks;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Purchase/PurchaseOrderDetail.cs
namespace ProductionManagement.Domain.Models.Purchase;

/// <summary>
/// 発注明細データ
/// </summary>
public class PurchaseOrderDetail
{
    public int Id { get; set; }
    public required string PurchaseOrderNumber { get; set; }
    public required int LineNumber { get; set; }
    public string? OrderNumber { get; set; }
    public string? DeliveryLocationCode { get; set; }
    public required string ItemCode { get; set; }
    public bool MiscellaneousItemFlag { get; set; }
    public required DateOnly ExpectedReceivingDate { get; set; }
    public DateOnly? ConfirmedDeliveryDate { get; set; }
    public required decimal OrderUnitPrice { get; set; }
    public required decimal OrderQuantity { get; set; }
    public decimal ReceivedQuantity { get; set; }
    public decimal InspectedQuantity { get; set; }
    public decimal AcceptedQuantity { get; set; }
    public required decimal OrderAmount { get; set; }
    public decimal TaxAmount { get; set; }
    public bool CompletedFlag { get; set; }
    public string? DetailRemarks { get; set; }
    public DateTime? CreatedAt { get; set; }
    public string? CreatedBy { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string? UpdatedBy { get; set; }
}
